File: backend/app/schemas/provider.py
from pydantic import BaseModel


class ProviderCapabilities(BaseModel):
    raster_generation: bool
    image_editing: bool
    region_editing: bool


class ProviderDescriptor(BaseModel):
    provider_name: str
    model_name: str | None
    available: bool
    capabilities: ProviderCapabilities
    note: str | None = None

File: frontend/src/types/provider.ts
export interface ProviderCapabilities {
  raster_generation: boolean
  image_editing: boolean
  region_editing: boolean
}

export interface ProviderDescriptor {
  provider_name: string
  model_name: string | null
  available: boolean
  capabilities: ProviderCapabilities
  note: string | null
}
